### DISCUSSION TITLE
[Feature] Hilfesystem

### DISCUSSION THREAD

**Discussion by AMatutat** [Programmiermethoden/Dungeon#1377] [Ideas]
In Spielen gibt es oft Hinweise für den Spieler, wenn dieser nicht weiterkommt. 
Im Kontext des Dungeon soll eine Hilfesystem eingebaut werden, welches dem Spieler Tipps für die Lösung der Rätsel gibt. 
Damit die Hinweise auch wirklich helfen, sollen diese kontextsensitive implementiert sein.

Es wird vermutlich kaum möglich sein, alle Problemstellen automatisch zu erkennen und dafür gute Hinweise zu geben. Aus der Erfahrung mit klassischen Prüfungsaufgaben konnte aber gelernt werden, die typischen Stolpersteine schon vor der Bearbeitung durch die Studierende zu erkennen (Vorzeichen Fehler, Klammer vergessen, etc.).

Diese Fälle sollen in der Aufgabenbeschreibung in der DSL markiert werden, mit einen manuell geschriebenen Hinweis versehen und, wenn der Spieler dann im Spiel an dieser Aufgabe (und dieser Stelle) scheitert, angezeigt werden.

**Comment by AMatutat** [comment]
Haben ein Petri Netz #2351

**Reply by cagix** [reply]
hmmm. ist ein hilfesystem nicht viel mehr? also sachen im dungeon anfassen und untersuchen (eigenschaften anzeigen)? das situationsabhängige anzeigen kriegt man über ein petrinetz hin, aber bisher fehlt doch die möglichkeit, überhaupt was zu untersuchen?
